File: LeetCode/1/1.cpp
#include<iostream>
#include<stdio.h>
#include<vector>
#include<map>
using namespace std;

vector<int> twoSum(vector<int>& nums, int target) {
    map<int,int> data;
    vector<int> ans;
    for(int i = 0; i < nums.size(); i++){
        if(data.find(target - nums[i]) != data.end()){
            cout << i << " " << data[target - nums[i]]<<endl;
            ans.push_back(i);
            ans.push_back(data[target - nums[i]]);
            break;
        }
        data[nums[i]] = i;
    }
    return ans;
}

int main()
{
    vector<int> vect { 3,2,4 };
    twoSum(vect, 6);
    vector<int> vect1 {3, 3};
    twoSum(vect1,6);
    return 0;
}

File: LeetCode/1/1.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer>data = new HashMap<>();
        int[] ans = new int[2];
        for(int i = 0; i < nums.length; i++) {
            int tmp = target - nums[i];
            if(data.containsKey(tmp)){
                System.out.println(data.get(tmp) + " " + i);
                ans[0] = data.get(tmp);
                ans[1] = i;
            }
            data.put(nums[i],i);
        }
        return ans;
    }

    public static void main(String[] args) {
        int[] tmp;
        tmp = twoSum(new int[] {3,2,4}, 6);
    }
}